### PR TITLE
Feature/query multiple tables

### PR DESCRIPTION
**Pull request by over-geek**
### Vet Clinic Database: Querying Multiple Tables

In this milestone, I:
- Created a table named owners with the following columns:
   - id: integer (set it as autoincremented PRIMARY KEY)
   - full_name: string
   - age: integer
- Created a table named species with the following columns:
   - id: integer (set it as autoincremented PRIMARY KEY)
   - name: string
- Modified animals table:
   - Made sure that id is set as autoincremented PRIMARY KEY
   - Removed column species
   - Added column species_id which is a foreign key referencing species table
   - Added column owner_id which is a foreign key referencing the owners table

- Inserted the following data into the owners table:
     - Sam Smith 34 years old.
     - Jennifer Orwell 19 years old.
     - Bob 45 years old.
     - Melody Pond 77 years old.
     - Dean Winchester 14 years old.
     - Jodie Whittaker 38 years old.
- Inserted the following data into the species table:
     - Pokemon
     - Digimon
- Modifed your inserted animals so it includes the species_id value:
     - If the name ends in "mon" it will be Digimon
     - All other animals are Pokemon
- Modified your inserted animals to include owner information (owner_id):
     - Sam Smith owns Agumon.
     - Jennifer Orwell owns Gabumon and Pikachu.
     - Bob owns Devimon and Plantmon.
     - Melody Pond owns Charmander, Squirtle, and Blossom.
     - Dean Winchester owns Angemon and Boarmon.

- [ ] Wrote queries to answer questions
Screenshot 2023-08-18 154238
Screenshot 2023-08-18 154341
Screenshot 2023-08-18 154457
Screenshot 2023-08-18 154529
Screenshot 2023-08-18 154555
Screenshot 2023-08-18 154612

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -41,4 +41,38 @@ SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
...
+
+SELECT animals.name, species.name AS species FROM animals
+  JOIN owners ON animals.owner_id = owners.id
+  JOIN species ON animals.species_id = species.id
+  WHERE owners.full_name = 'Melody Pond';
+
+SELECT animals.name, owners.full_name FROM animals
+  JOIN owners ON animals.owner_id = owners.id
+  JOIN species ON animals.species_id = species.id
+  WHERE species.name = 'Pokemon';
```
Comment (muneebulrehman) [inline queries.sql]: The results you attached for the query are not showing the right results. It is showing **Pikachu**  two times belonging to the same person. Kindly fix this issue.

File: queries.sql [hunk 2]
```diff
@@ -41,4 +41,38 @@ SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
...
+
+SELECT species.name AS species_name, COUNT(*) AS num_animals FROM animals
+  JOIN species ON animals.species_id = species.id
+  GROUP BY species.id, species.name
+  ORDER BY num_animals DESC;
+
+SELECT animals.name AS animal_name, species.name AS species_name, owners.full_name FROM animals
+  JOIN species ON animals.species_id = species.id
+  JOIN owners ON animals.owner_id = owners.id
+  WHERE species.name = 'Digimon' AND owners.full_name = 'Jennifer Orwell';
```
Comment (muneebulrehman) [inline queries.sql]: The results you have attached for this query don't return the right results. It is returning **Gabumon** two times. Kindly fix this issue.

File: queries.sql [hunk 3]
```diff
@@ -41,4 +41,38 @@ SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
...
+
+SELECT name AS non_esc_animals FROM animals
+  INNER JOIN owners ON animals.owner_id = owners.id
+  WHERE animals.owner_id = (SELECT id FROM owners WHERE full_name = 'Dean Winchester') AND animals.escape_attempts = 0;
+
+SELECT owners.full_name, COUNT(*) AS num_animals_owned FROM animals
+  JOIN owners ON animals.owner_id = owners.id
+  GROUP BY owners.full_name
+  ORDER BY num_animals_owned DESC LIMIT 1;
```
Comment (muneebulrehman) [inline queries.sql]: As per the results you have attached for this query, it is not returning the correct results. The owner of most animals is **Melody Pond**. However, your query is returning **Jennifer Orwell**. Kindly fix this issue.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @over-geek , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR
- Data inserted correctly
- Added new table and columns.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mckent05 — APPROVED** [review]
Hi @over-geek, 

# STATUS: APPROVED ✔️ ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations and well done! 🎉 
approved


## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👏 👏 
- Good job attaching the right results 👍 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**